widgetastic_manageiq/__init__.py

```diff
@@ -4356,3 +4356,81 @@ def fill(self, value):
             return False
         self._all_item_elements[value or self.selected_item].click()
         return True
+
+
+class MigrationPlansList(Widget):
+    """Represents the list of Migration Plans."""
+
+    ROOT = ParametrizedLocator('.//div[contains(@class,{@list_class|quote})]')
+    ITEM_LOCATOR = ParametrizedLocator('.//div[contains(@class,"{@list_class}__list-item")]')
+    # ITEM_TEXT_LOCATOR helps locate name of the Migration plan. ITEM_LOCATOR.text does not suffice.
+    # Also, ITEM_LOCATOR does not yield element which can be clicked to navigate to details page.
+    ITEM_TEXT_LOCATOR = './/div[contains(@class,"list-group-item-heading")]'
+    ITEM_TIMER_LOCATOR = './/div[./span[contains(@class,"fa-clock-o")]]'
+    ITEM_DESCRIPTION_LOCATOR = './/div[contains(@class,"list-group-item-text")]'
+    ITEM_VM_LOCATOR = ('.//div[./span[contains(@class,"pficon-virtual-machine") '
+            'or contains(@class,"pficon-screen")]]')
+    ITEM_BUTTON_LOCATOR = './div[contains(@class,"list-view-pf-actions")]//button'
+
+    def __init__(self, parent, list_class, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.list_class = list_class
+
+    @property
+    def all_items(self):
+        return [self.browser.text(item) for item in self.browser.elements(self.ITEM_TEXT_LOCATOR)]
+
+    def get_vm_count_in_plan(self, plan_name):
+        el = self._get_plan_element(plan_name)
+        return self.browser.text(self.browser.element(self.ITEM_VM_LOCATOR, parent=el))
+
+    def get_clock(self, plan_name):
+        try:
+            el = self._get_plan_element(plan_name)
+            return self.browser.text(self.browser.element(self.ITEM_TIMER_LOCATOR, parent=el))
+        except NoSuchElementException:
+            # Plan with no clock
+            return None
+
+    def get_plan_description(self, plan_name):
+        try:
+            el = self._get_plan_element(plan_name)
+            return self.browser.text(self.browser.element(self.ITEM_DESCRIPTION_LOCATOR,
+                 parent=el))
+        except NoSuchElementException:
+            # Plan with no description
+            return None
+
+    def _get_plan_element(self, plan_name):
+        for item in self.browser.elements(self.ITEM_LOCATOR):
+            el = self.browser.element(".//*[@class='list-group-item-heading']", parent=item)
+            if self.browser.text(el) == plan_name:
+                return item
+        raise ItemNotFound("Migration Plan: {} not found".format(plan_name))
```

Comment:
when I pass a non-existing plan name to any of the methods, ideally it should not do anything or throw exception, but it was not doing so. It was performing the asked operation on first item in the list if plan_name that I passed does not exist in the list. Hence to avoid that I added exception here, so for any non-existent plan_name it will throw exception and will not perform any operations.